### PR TITLE
Fix addThingView bug

### PR DESCRIPTION
**Pull request by alexxcheung**
Add ActionsheetManager

### CODE REVIEW

File: LetsDinnerV2 MessagesExtension/View/CustomUIView/SectionSelectionInput.swift [hunk 1]
```diff
@@ -79,8 +80,12 @@ class SectionSelectionInput : UIView {
         // Maybe adding constraints in layoutSubviews fix the bug of the toolnar not always appearing
         addConstraints()
         
-        // Control the selected bubble (First time)
-        sectionsCollectionView.selectItem(at: [0,0], animated: true, scrollPosition: .top)
+        if viewFirstInit {
+            // Show the selected bubble (First time)
+            sectionsCollectionView.selectItem(at: [0,0], animated: true, scrollPosition: .top)
+            viewFirstInit = false
```
Comment (Ericordo) [inline LetsDinnerV2 MessagesExtension/View/CustomUIView/SectionSelectionInput.swift:86]: Really? There is no other way? That looks very dodgy

File: LetsDinnerV2 MessagesExtension/Controller/RegistrationVC/RegistrationViewController.swift [hunk 2]
```diff
@@ -537,6 +523,7 @@ class RegistrationViewController: LDNavigationViewController {
             make.top.equalToSuperview().offset(10)
             make.width.equalTo(userPic.snp.height)
             make.bottom.equalTo(addPicButton.snp.top).offset(-10)
+            make.centerY.equalToSuperview()
```
Comment (Ericordo) [inline LetsDinnerV2 MessagesExtension/Controller/RegistrationVC/RegistrationViewController.swift]: Why? There is already a top and a bottom constraint

File: LetsDinnerV2 MessagesExtension/Extensions/UIImage.swift [hunk 3]
```diff
@@ -18,4 +18,17 @@ extension UIImage {
         guard let outputImage = filter.outputImage else { return nil }
         return UIImage(ciImage: outputImage)
     }
+    
+    func withColor(_ color: UIColor) -> UIImage {
```
Comment (Ericordo) [inline LetsDinnerV2 MessagesExtension/Extensions/UIImage.swift]: Do you just use this to change the color of the image?

File: LetsDinnerV2 MessagesExtension/Manager/ActionSheetManager.swift [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+//  ActionSheetManager.swift
+//  LetsDinnerV2 MessagesExtension
+//
+//  Created by Alex Cheung on 1/7/2020.
+//  Copyright © 2020 Eric Ordonneau. All rights reserved.
+//
+
+import UIKit
+
+class ActionSheetManager {
```
Comment (Ericordo) [inline LetsDinnerV2 MessagesExtension/Manager/ActionSheetManager.swift:11]: No you can't do that, this is all but generic. You just put action sheets in a single class? And then each method is specific to just one place in the app? What is the point? You remove all the privacy. Then why not put all the labels in a label manager class, the button in a button manager class? This stuff means i can call presentSaveActionSheetInRegistrationVC in any view controller. This is bad

**Comment by Ericordo** [comment]
Don't call a branch feature if it is a bugfix

File: LetsDinnerV2 MessagesExtension/View/CustomUIView/SectionSelectionInput.swift (hunk 1 above)
Comment (Ericordo) [reply LetsDinnerV2 MessagesExtension/View/CustomUIView/SectionSelectionInput.swift:86]: Actually the bug does not seem to be there on 1.0.25 so no need for this

Comment (alexxcheung) [reply LetsDinnerV2 MessagesExtension/View/CustomUIView/SectionSelectionInput.swift:86]: the bug is: when hiding the keyboard, if the selectedSection are saved at the second item, but the bubble appears at Misc when showing the keyboard again

File: LetsDinnerV2 MessagesExtension/Controller/RegistrationVC/RegistrationViewController.swift (hunk 2 above)
Comment (alexxcheung) [reply LetsDinnerV2 MessagesExtension/Controller/RegistrationVC/RegistrationViewController.swift]: mistake

File: LetsDinnerV2 MessagesExtension/Extensions/UIImage.swift (hunk 3 above)
Comment (alexxcheung) [reply LetsDinnerV2 MessagesExtension/Extensions/UIImage.swift]: yes

File: LetsDinnerV2 MessagesExtension/Manager/ActionSheetManager.swift (hunk 4 above)
Comment (alexxcheung) [reply LetsDinnerV2 MessagesExtension/Manager/ActionSheetManager.swift:11]: no you can only call it from registrationVC

**Comment by alexxcheung** [comment]
Coz Gitflow has no bugfix branch.

File: LetsDinnerV2 MessagesExtension/Manager/ActionSheetManager.swift (hunk 4 above)
Comment (alexxcheung) [reply LetsDinnerV2 MessagesExtension/Manager/ActionSheetManager.swift:11]: Generalized

**Comment by Ericordo** [comment]
What do you mean? You already created bugFix branches before

File: LetsDinnerV2 MessagesExtension/Extensions/UIImage.swift (hunk 3 above)
Comment (Ericordo) [reply LetsDinnerV2 MessagesExtension/Extensions/UIImage.swift]: Is this not working?
theImageView.image = theImageView.image?.withRenderingMode(.alwaysTemplate)
theImageView.tintColor = UIColor.red
https://stackoverflow.com/questions/19274789/how-can-i-change-image-tintcolor-in-ios-and-watchkit

File: LetsDinnerV2 MessagesExtension/Controller/RecipeCreationVC/RecipeCreationViewController.swift [hunk 5]
```diff
@@ -598,9 +607,8 @@ class RecipeCreationViewController: UIViewController  {
     }
     
     private func createDefaultImage() -> UIImage {
-        let imageName = "emptyPlate"
-        let image = UIImage(named: imageName)
-        return image!
+        let image = Images.emptyPlate
+        return image
```
Comment (Ericordo) [inline LetsDinnerV2 MessagesExtension/Controller/RecipeCreationVC/RecipeCreationViewController.swift]: What is the point of this method? Just assign Images.emptyPlate to recipeImage

**Comment by alexxcheung** [comment]
but bugfix branch should not be existed

File: LetsDinnerV2 MessagesExtension/Extensions/UIImage.swift (hunk 3 above)
Comment (alexxcheung) [reply LetsDinnerV2 MessagesExtension/Extensions/UIImage.swift]: Both works.

**Comment by Ericordo** [comment]
you don't need to respect gitflow

**Review by Ericordo — APPROVED** [review]

**Comment by alexxcheung** [comment]
I only respect you, not gitflow

**Pull request merged**